WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+        ]
+    }
+
+    // MARK: - Styling Options
+
+    private lazy var optionsTablePresenter = OptionsTablePresenter(presentingViewController: self, presentingTextView: editorView.richTextView)
+
+    /// Aztec's Awesomeness
+    ///
+    private var richTextView: Aztec.TextView {
+        get {
+            return editorView.richTextView
+        }
+    }
+
+    /// Raw HTML Editor
+    ///
+    private var htmlTextView: UITextView {
+        get {
+            return editorView.htmlTextView
+        }
+    }
+
+    /// Aztec's Text Placeholder
+    ///
+    fileprivate(set) lazy var placeholderLabel: UILabel = {
+        let label = UILabel()
+        label.text = NSLocalizedString("Start writing...", comment: "Aztec's Text Placeholder")
+        label.textColor = StyleManager.wooGreyMid
+        label.font = StyleManager.subheadlineFont
+        label.isUserInteractionEnabled = false
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.textAlignment = .natural
+        label.accessibilityIdentifier = "aztec-content-placeholder"
+        return label
+    }()
+
+    /// Current keyboard rect used to help size the inline media picker
+    ///
+    fileprivate var currentKeyboardFrame: CGRect = .zero
+
+    required init(content: String?) {
+        self.content = content ?? ""
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        registerAttachmentImageProviders()
+
+        configureNavigationBar()
+        configureView()
+        configureSubviews()
+
+        configureConstraints()
+
+        setHTML(content)
+
+        refreshPlaceholderVisibility()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        startListeningToNotifications()
+    }
+
+    deinit {
+        stopListeningToNotifications()
+    }
+}
+
+private extension AztecEditorViewController {
+    func configureNavigationBar() {
+        title = NSLocalizedString("Description", comment: "The navigation bar title of the Product description editor screen.")
+
+        navigationController?.navigationBar.isTranslucent = false
+        navigationController?.navigationBar.accessibilityIdentifier = "Aztec Editor Navigation Bar"
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(saveButtonTapped))
+    }
+
+    func configureView() {
+        edgesForExtendedLayout = UIRectEdge()
+        view.backgroundColor = StyleManager.wooWhite
+    }
+
+    func configureSubviews() {
+        view.addSubview(richTextView)
+        view.addSubview(htmlTextView)
+        view.addSubview(placeholderLabel)
+    }
+
+    private func setupHTMLTextView(_ textView: UITextView) {
```

Comment:
There are several methods that setup views, and that receive a view as a parameter. One way to make this vc lighter might be extracting those methods, and all other properties / helper methods that are associated to them to a separate something (it might be a class, or just a struct that provides sort of a namespace for those methods).

That has the potential or removing lots of code from this VC, and most importantly, remove responsibilities from it (responsibilities that could be potentially mocked / injected if necessary)